Networking/ServerConnection.cs

```diff
@@ -45,26 +45,25 @@ public override void _Ready()
 
     _serilogger.Debug("ServerConnection.cs: Attempting to load embedded client config");
     err = clientConfig.Load("res://Resources/client.cfg");
-    if (err == Godot.Error.Ok)
-    {
-      _serilogger.Information("ServerConnection.cs: Successfully loaded the config from 'res://Resources/client.cfg'");
-      url = (String)clientConfig.GetValue("amqp", "server_string", "amqp://127.0.0.1:5672");
-      _serilogger.Debug("ServerConnection.cs: config file: setting url to " + url);
-      disableCertValidation = (bool)clientConfig.GetValue("amqp", "disable_cert_validation", true);
-      _serilogger.Debug("ServerConnection.cs: config file: setting cert validation to " + disableCertValidation);
-    }
-
-    _serilogger.Debug("ServerConnection.cs: Overriding with client local/user config");
-    err = clientConfig.Load("user://client.cfg");
-    if (err == Godot.Error.Ok)
+    if (err != Godot.Error.Ok)
     {
+      err = clientConfig.Load("user://client.cfg");
       _serilogger.Information("ServerConnection.cs: Successfully loaded the config from 'user://client.cfg'");
-      url = (String)clientConfig.GetValue("amqp", "server_string", "amqp://127.0.0.1:5672");
-      _serilogger.Debug("ServerConnection.cs: config file: setting url to " + url);
-      disableCertValidation = (bool)clientConfig.GetValue("amqp", "disable_cert_validation", true);
-      _serilogger.Debug("ServerConnection.cs: config file: setting cert validation to " + disableCertValidation);
+
+      // If config file failed to load and the game is in debug mode throw an error
+      if (err != Godot.Error.Ok && OS.IsDebugBuild())
+      {
+        throw new Exception("ServerConnection.cs: Failed to load the client configuration file");
+      }
     }
 
+    url = OS.GetEnvironment("SERVER_STRING") ?? (String)clientConfig.GetValue("amqp", "server_string", "amqp://127.0.0.1:5672");
```

Comment:
I don't consider the case where the required environment variables are not present, in production